numpy/random/_mt19937.pyx

```diff
@@ -129,7 +129,7 @@ cdef class MT19937(BitGenerator):
         BitGenerator.__init__(self, seed)
         val = self._seed_seq.generate_state(RK_STATE_LEN, np.uint32)
         # MSB is 1; assuring non-zero initial array
-        self.rng_state.key[0] = 0x80000000UL
+        self.rng_state.key[0] |= 0x80000000UL
         for i in range(1, RK_STATE_LEN):
             self.rng_state.key[i] = val[i]
```

Comment:
Check this against the logic in the original code. All of the values must be copied in first, _then_ the MSB gets ORed to 1.

```
for i in range(RK_STATE_LEN):
    self.rng_state.key[i] = val[i]
self.rng_state.key[0] |= 0x80000000UL
```

Reply:
fixed